### ISSUE TITLE
Moving cursor to before an Import causes server to lose connection, does not recover.

### ISSUE THREAD

**Issue by collin-marando** [coq-community/vscoq#701]
I am using VSCoq 2, with manual proof mode. When stepping through a document, if I then decide to click anywhere before an Import clause, the server crashes with the following error dump:

```
[Info  - 5:38:54 PM] Connection to server got closed. Server will restart.
[Error - 5:38:54 PM] Request textDocument/hover failed.
Error: Connection got disposed.
    at Object.dispose (/Users/co6101552/.vscode/extensions/maximedenes.vscoq-2.0.3/dist/extension.js:1:78437)
    at Object.dispose (/Users/co6101552/.vscode/extensions/maximedenes.vscoq-2.0.3/dist/extension.js:1:158523)
    at s.handleConnectionClosed (/Users/co6101552/.vscode/extensions/maximedenes.vscoq-2.0.3/dist/extension.js:1:158736)
    at s.handleConnectionClosed (/Users/co6101552/.vscode/extensions/maximedenes.vscoq-2.0.3/dist/extension.js:1:219089)
    at t (/Users/co6101552/.vscode/extensions/maximedenes.vscoq-2.0.3/dist/extension.js:1:156825)
    at r.invoke (/Users/co6101552/.vscode/extensions/maximedenes.vscoq-2.0.3/dist/extension.js:1:80050)
    at o.fire (/Users/co6101552/.vscode/extensions/maximedenes.vscoq-2.0.3/dist/extension.js:1:80811)
    at Q (/Users/co6101552/.vscode/extensions/maximedenes.vscoq-2.0.3/dist/extension.js:1:67695)
    at r.invoke (/Users/co6101552/.vscode/extensions/maximedenes.vscoq-2.0.3/dist/extension.js:1:80050)
    at o.fire (/Users/co6101552/.vscode/extensions/maximedenes.vscoq-2.0.3/dist/extension.js:1:80811)
    at m.fireClose (/Users/co6101552/.vscode/extensions/maximedenes.vscoq-2.0.3/dist/extension.js:1:88692)
    at Socket.<anonymous> (/Users/co6101552/.vscode/extensions/maximedenes.vscoq-2.0.3/dist/extension.js:1:90277)
    at Socket.emit (node:events:525:35)
    at Pipe.<anonymous> (node:net:322:12)
```

The server does not restart, and the proof highlighting remains but will no longer update. vscoqtop cannot be terminated in the Activity Monitor. The only known solution is to restart VSCode.

**Comment by rtetley** [comment]
Hi ! Do you have a minimal example ? While trying to reproduce for me I get the same crash when I hover over the first word of the document but not in the situation you describe !

**Comment by rtetley** [comment]
I'm closing this as I think it was solved by #721. Feel free to re-open if this wasn't corrected.

**Comment by collin-marando** [comment]
I think you may be right. I could very well be mistaken about it happening anywhere, and not just on the first line. I'll keep an eye out, but this certainly seems to have solved it for now. Thanks!
